lib/ret_web/channels/hub_channel.ex

```diff
@@ -249,15 +312,23 @@ defmodule RetWeb.HubChannel do
         socket
       ) do
     with_account(socket, fn account ->
-      perform_pin!(object_id, gltf_node, account, socket)
-      Storage.promote(file_id, file_access_token, promotion_token, account)
-      OwnedFile.set_active(file_id, account.account_id)
+      hub = socket |> hub_for_socket
+
+      if account |> can?(pin_objects(hub)) do
```

Comment:
Will we need `can?(unpin_objects(hub))` below in `handle_in("unpin" ...`?

Reply:
Nice catch! Thanks.

Reply:
oops, missed this!